src/pks/flow/constitutive_relations/sources/field_mass_source_evaluator.cc

```diff
@@ -0,0 +1,69 @@
+/*
+  Copyright 2010-202x held jointly by participating institutions.
+  ATS is released under the three-clause BSD License.
+  The terms of use and "as is" disclaimer for this license are
+  provided in the top-level COPYRIGHT file.
+
+  Authors: Phong V.V. Le
+*/
+
+#include "Key.hh"
+#include "Factory.hh"
+#include "Function.hh"
+#include "field_mass_source_evaluator.hh"
+#include "FunctionFactory.hh"
+
+namespace Amanzi {
+namespace Flow {
+namespace Relations {
+
+
+FieldMassSourceEvaluator::FieldMassSourceEvaluator(Teuchos::ParameterList& plist) : EvaluatorSecondaryMonotypeCV(plist)
+{
+  domain_ = Keys::getDomain(my_keys_.front().first);
+  auto tag = my_keys_.front().second;
+
+  cv_key_ = Keys::readKey(plist, domain_, "cell volume", "cell_volume");
+  dependencies_.insert(KeyTag{ cv_key_, tag });
+  molar_density_key_ = Keys::readKey(plist, domain_, "molar density liquid", "molar_density_liquid");
+  dependencies_.insert(KeyTag{ molar_density_key_, tag });
+  field_src_key_ = Keys::readKey(plist, domain_, "field mass source", "field_mass_source");
```

Comment:
So I've switched to mostly calling these "water_source" rather than "mass_source".  Mass could be mass of transported stuff, or it could be mass of water.